yarn-project/noir-contracts/src/contracts/easy_private_voting_contract/src/main.nr

```diff
@@ -0,0 +1,117 @@
+contract EasyPrivateVoting {
+    // docs:start:imports
+    use dep::aztec::{
+	context::{PrivateContext, Context},
+    oracle::get_secret_key::get_secret_key, // used to compute nullifier
+    selector::compute_selector, // used to compute function selector for calling a function 
+	 state_vars::{ map::Map, public_state::PublicState,}, 
+     types::type_serialization::{ // serialization methods for using booleans and aztec addresses
+	    bool_serialization::{BoolSerializationMethods, BOOL_SERIALIZED_LEN},
+	    aztec_address_serialization::{AztecAddressSerializationMethods, AZTEC_ADDRESS_SERIALIZED_LEN},
+        field_serialization::{ FieldSerializationMethods, FIELD_SERIALIZED_LEN},
+	}, 
+    types::address::{AztecAddress},
+    constants_gen::EMPTY_NULLIFIED_COMMITMENT,
+    };
+    // docs:end:imports
+    // docs:start:storage_struct
+    struct Storage {
+    admin: PublicState<AztecAddress, AZTEC_ADDRESS_SERIALIZED_LEN>, // admin can end vote
+    tally: Map<PublicState<Field, FIELD_SERIALIZED_LEN>>, // we will store vote as key and number of votes as value
+    voteEnded: PublicState<bool, BOOL_SERIALIZED_LEN>, // voteEnded is boolean
+}
+    // docs:end:storage_struct
+    // docs:start:storage_impl
+    impl Storage {
+	fn init(context: Context) -> Self {
+	    Storage {
+		 admin: PublicState::new(
+		    context, 
+            1, // storage slot. this can be anything except 0. it is hashed, and hash on 0 = 0
+            AztecAddressSerializationMethods,
+		), 
+        tally: Map::new(
+		    context, 
+            2, 
+            |context, slot| {
+			PublicState::new(
+			    context,
+                slot, 
+                FieldSerializationMethods,
+			)
+		    },
+		 ),
+		  voteEnded: PublicState::new(
+		    context, 
+            3, 
+            BoolSerializationMethods,
+		)
+	    } }
+	}
+    // docs:end:storage_impl
+    // docs:start:constructor
+    #[aztec(private)] // annotation to mark function as private and expose private context
+    fn constructor(admin: AztecAddress) { // called when contract is deployed
+        context.call_public_function(
+            // we cannot update public state directly from private function but we can call public function (which queues it)
+            context.this_address(),// address that calls public function. here we are using contract address for privacy
```

Comment:
comment is misleading.
You don't define which address calls public function. This parameter is the contract address whose function you want to call.

`// contract address whose method we want to call`